packages/next-codemod/transforms/lib/async-request-api/utils.ts

```diff
@@ -492,3 +492,66 @@ export function getVariableDeclaratorId(
   }
   return undefined
 }
+
+export function findFunctionBody(path: ASTPath<FunctionScope>): null | any[] {
+  let functionBody = path.node.body
+  if (functionBody && functionBody.type === 'BlockStatement') {
+    return functionBody.body
+  }
+  return null
+}
+
+export function containsReactHooksCallExpressions(
+  path: ASTPath<FunctionScope>,
+  j: API['jscodeshift']
+) {
+  const hasReactHooks =
+    j(path)
+      .find(j.CallExpression)
+      .filter((callPath) => {
+        return (
+          j.Identifier.check(callPath.value.callee) &&
+          callPath.value.callee.name.startsWith('use')
+        )
+      })
+      .size() > 0
+  return hasReactHooks
+}
+
+export function isParentUseCallExpression(
+  path: ASTPath<any>,
+  j: API['jscodeshift']
+) {
+  if (
+    // member access parentPath is argument
+    j.CallExpression.check(path.parent.value) &&
+    // member access is first argument
+    path.parent.value.arguments[0] === path.value &&
+    // function name is `use`
+    j.Identifier.check(path.parent.value.callee) &&
+    path.parent.value.callee.name === 'use'
+  ) {
+    return true
+  }
+  return false
+}
```

Comment:
Can you add some examples what this detects? This is hard to follow with the current factoring.

Reply:
I didn't refactor the inside, I mostly just moved around the utils from the codemod to the utils

Reply:
And when you move them to be usable utils from all places, you should make them understandable. Otherwise keep them private.